pandas/io/excel/_xlsxwriter.py

```diff
@@ -248,3 +283,45 @@ def write_cells(
                 )
             else:
                 wks.write(startrow + cell.row, startcol + cell.col, val, style)
+
+    def _value_with_fmt(
+        self, val: Any, col: Optional[int] = None
+    ) -> Tuple[Any, Optional[str]]:
+        """
+        Convert numpy types to Python types for the Excel writer.
+
+        Parameters
+        ----------
+        val : object
+            Value to be written into cells.
+        col : int, optional
+            Column number of the cell.
+
+        Returns
+        -------
+        Tuple with the first element being the converted value and the second
+            being an optional format
+        """
+        col_format = None
+        fmt = None
+
+        if self.num_formats:
+            col_format = self.num_formats.get(col, None)
+
+        if is_integer(val):
+            val = int(val)
+        elif is_float(val):
```

Comment:
this is way too general for just a numeric formatter. i suppose this could be called `formatters` instead, but then would want to deprecate `date_format`

Reply:
This method is overriden, I think I will modify in the base itself.